test/src/main/java/org/apache/accumulo/test/functional/MemoryStarvedScanIT.java

```diff
@@ -392,21 +489,28 @@ public void testBatchScanPauses() throws Exception {
         assertEquals(currentCount, fetched.get());
         assertTrue(SCAN_START_DELAYED.doubleValue() >= paused);
         assertEquals(returned, SCAN_RETURNED_EARLY.doubleValue());
-        assertEquals(1, LOW_MEM_DETECTED.get());
 
+        // check across multiple low memory checks and metric updates that low memory detected
+        // remains set
+        int checkCount = 6;
+        while (checkCount-- > 0) {
```

Comment:
This currently runs 6 times (5, 4, 3, 2, 1, 0). It seems like you intended it to run 5 times (5, 4, 3, 2, 1).

```suggestion
        while (--checkCount > 0) {
```

You could also be more explicit about the range you want to iterate over by using IntStream and forEach:

```java
IntStream.rangeClosed(1, 5).map(i -> 6-i).forEach(System.out::println); // or
IntStream.rangeClosed(-5, -1).map(i -> -i).forEach(System.out::println);
```

That example goes backwards, like yours, but it's easier to go forwards.

Reply:
There is no requirement for number of iterations. The more iterations the greater the confidence that the value is not flapping. So, there was a trade-off of greater confidence vs increased test time. 

At a minimum, for the test to be valid the metrics do need to update at least once.

Changed it to count up in 3ed08fc601, mainly for logging. Chose not to use IntStream here because it seems harder to understand for a common loop construct in a test,